lib/item_marketplace/local_repository/local_item_repository.rb

```diff
@@ -0,0 +1,42 @@
+require_relative 'local_item'
+
+module LocalRepository
+  class ItemRepository
+    def initialize
+      @items = {}
+      @id = 1
+    end
+
+    def model_class
+      LocalRepository::LocalItem
+    end
+
+    def new_item(item_details = {})
+      model_class.new(item_details)
+    end
+
+    def save(item)
+      item.id = @id
+      if  item.valid?
+        @items[@id] = item
```

Comment:
A repository should not have validation logic in it. A repository just transfers data from the app to db and from db to the app.